src/lein_monolith/task/with_dependency_set.clj

```diff
@@ -0,0 +1,55 @@
+(ns lein-monolith.task.with-dependency-set
+  (:require
+    [clojure.java.io :as io]
+    [lein-monolith.plugin :as plugin]
+    [lein-monolith.task.util :as u]
+    [leiningen.core.main :as lein]
+    [leiningen.core.project :as project]))
+
+
+(defn- maybe-remove-dependencies
+  [project opts]
+  (if (:only opts)
+    (assoc project :dependencies '())
```

Comment:
Use an empty vector to preserve the type, or better dissociate the key entirely.
```suggestion
    (dissoc project :dependencies)
```

Reply:
I may've misread it and will take a look but I think the `:dependencies` get evaluated as a list and not a vector interestingly enough. I think preserving an empty seq is important for some downstream tasks that expect a non-nil value for this. I'll do a little more digging on that though.

Reply:
Okay, totally fine leaving this if we need a non-nil value, though I think a vector would still be better because that's what is actually in `project.clj` files. Clojure treats lists and vectors as belonging to the same "equality class", I think it's probably showing up as a sequence in `pprint` outputs because the dependencies from profiles get concatenated with the main dependency vector as they are merged in.